app/scripts/modules/core/pipeline/config/stages/executionWindows/ExecutionWindowDayPickerComponent.js

```diff
@@ -0,0 +1,89 @@
+import angular from 'angular';
+
+class ExecutionWindowDayPickerController {
+
+  constructor() {
+    this.DAYS = [{
+      dow: 'sunday',
+      abbr: 'Sun'
+    }, {
+      dow: 'monday',
+      abbr: 'Mon'
+    }, {
+      dow: 'tuesday',
+      abbr: 'Tue'
+    }, {
+      dow: 'wednesday',
+      abbr: 'Wed'
+    }, {
+      dow: 'thursday',
+      abbr: 'Thu'
+    }, {
+      dow: 'friday',
+      abbr: 'Fri'
+    }, {
+      dow: 'saturday',
+      abbr: 'Sat'
+    }];
+    this.dayModel = {};
+  }
+
+  $onInit() {
+    console.log('executing onInit');
+    const days = new Set(this.days);
+    this.DAYS.forEach((day, index) => this.dayModel[day.dow] = days.has(index));
+  }
+
+  all() {
+    this.dayModel.sunday = this.dayModel.saturday = this.dayModel.monday =
+      this.dayModel.tuesday = this.dayModel.wednesday =
+        this.dayModel.thursday = this.dayModel.friday = true;
+    this.updateModel();
+  }
+
+  none() {
+    this.dayModel.sunday = this.dayModel.saturday = this.dayModel.monday =
+      this.dayModel.tuesday = this.dayModel.wednesday =
+        this.dayModel.thursday = this.dayModel.friday = false;
+    this.updateModel();
+  }
+
+  weekdays() {
+    this.dayModel.sunday = this.dayModel.saturday = false;
+    this.dayModel.monday = this.dayModel.tuesday = this.dayModel.wednesday =
+      this.dayModel.thursday = this.dayModel.friday = true;
+    this.updateModel();
+  }
+
+  weekend() {
+    this.dayModel.sunday = this.dayModel.saturday = true;
+    this.dayModel.monday = this.dayModel.tuesday = this.dayModel.wednesday =
+      this.dayModel.thursday = this.dayModel.friday = false;
+    this.updateModel();
+  }
+
+  updateModel() {
+    const updated = [];
+    this.DAYS.forEach((day, index) => {
+      if (this.dayModel[day.dow]) {
+        updated.push(index + 1);
+      }
+    });
+    this.days = updated;
```

Comment:
@anotherchrisberry yes, this works and i've confirmed across page reloads that saved values are re-retrieved and set correctly.